File: impls/kotlin/src/mal/step0_repl.kt
fun READ(s: String) = s

fun EVAL(s: String) = s

fun PRINT(s: String) = println(s)

fun rep(s: String) = PRINT(EVAL(READ(s)))

fun main(args: Array<String>) {
        while(true) {
                print("user> ")
                readLine()?.let { rep(it) }
        }
}

File: impls/erlang/src/step0_repl.erl
-module(step0_repl).

-export([main/1]).

main(_) ->
    case io:get_line(standard_io, "user> ") of
        eof ->
            io:format("~n"),
            ok;
        {error, ErrorDescription} ->
            io:format("Error reading input: ~p~n", [ErrorDescription]),
            exit(ioerr);
        Input -> 
            io:format("~s~n", [Input]),
            main("")
        end.

read(Code) ->
    Code.

eval(Code) ->
    Code.

print(Code) ->
    Code.
